libbeat/kibana/client.go

```diff
@@ -92,6 +92,36 @@ func extractError(result []byte) error {
 	return nil
 }
 
+func extractMessage(result []byte) error {
+	var kibanaResult struct {
+		Success bool
+		Errors  []struct {
+			Id    string
+			Type  string
+			Error struct {
+				Type       string
+				References []struct {
+					Type string
+					Id   string
+				}
+			}
+		}
+	}
+	if err := json.Unmarshal(result, &kibanaResult); err != nil {
+		return nil
+	}
+
+	if !kibanaResult.Success {
+		var errs multierror.Errors
+		for _, err := range kibanaResult.Errors {
+			errs = append(errs, fmt.Errorf("error: %s, asset ID=%s; asset type=%s; references=%+v", err.Error.Type, err.Id, err.Type, err.Error.References))
```

Comment:
I guess that there can be other types of errors here, and other errors may not have anything to do with references, so `err.Error.Refereces` will be `nil`. But we can revisit this later if we find this is problematic, or we need to get more info for for other kinds of errors.

Reply:
The whole error parsing from Kibana is very fragile, so I agree. I have removed the previous error check because it is just going to break. It would be nice if we could get HTTP errors when errors do happen: e.g. missing references.